pkg/tool/kube/informer/informer.go

```diff
@@ -0,0 +1,70 @@
+package informer
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/koderover/zadig/pkg/setting"
+)
+
+var InformersMap sync.Map
+var StopChanMap sync.Map
+
+// NewInformer initialize and start an informer for specific namespace in given cluster.
+// Currently the informer will NOT stop unless the service is down
+// If you want to watch a new resource, remember to register here
+// Current list:
+// - Deployment
+// - StatefulSet
+// - Service
+// - Pod
+// - Ingress (extentions/v1beta1) <- as of version 1.9.0, this is the resource we watch
+func NewInformer(clusterID, namespace string, cls *kubernetes.Clientset) informers.SharedInformerFactory {
+	// this is a stupid compatibility code
+	if clusterID == "" {
+		clusterID = setting.LocalClusterID
+	}
+	key := fmt.Sprintf(setting.InformerNamingConvention, clusterID, namespace)
```

Comment:
Consider abstracting the naming rules generated by this key into a public method?